### PR TITLE
Nitzan/eco 752/logout

### PR DESCRIPTION
**Pull request by nitzantomer**
added a DELETE `users/me/session` endpoint

### CODE REVIEW

File: scripts/src/public/auth.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ async function getTokenAndUser(req: express.Request): Promise<[db.AuthToken, db.
 	}
 
 	const token = await db.AuthToken.findOneById(req.token);
-	if (!token) {
+	if (!token || !token.valid) {
```
Comment (doodyparizada) [inline scripts/src/public/auth.ts]: great catch. I wonder if clients will start failing or that the refresh mechanism will work @YohayBar ?

File: scripts/src/public/routes/users.ts [hunk 2]
```diff
@@ -127,3 +128,8 @@ export const myUserInfo = async function(req: Request, res: Response) {
 	req.params.user_id = req.context.user!.appUserId;
 	await (userInfo as any)(req as UserInfoRequest, res);
 } as any as RequestHandler;
+
+export const logoutUser = async function(req: Request, res: Response) {
+	await logoutService(req.context.user!, req.context.token!);
+	res.status(200).send();
```
Comment (doodyparizada) [inline scripts/src/public/routes/users.ts]: usually if there is no output we use 204 (No Content)

File: scripts/src/public/routes/users.ts [hunk 3]
```diff
@@ -127,3 +128,8 @@ export const myUserInfo = async function(req: Request, res: Response) {
 	req.params.user_id = req.context.user!.appUserId;
 	await (userInfo as any)(req as UserInfoRequest, res);
 } as any as RequestHandler;
+
+export const logoutUser = async function(req: Request, res: Response) {
+	await logoutService(req.context.user!, req.context.token!);
```
Comment (doodyparizada) [inline scripts/src/public/routes/users.ts]: I was thinking that we should use a different `Request` object for authenticated and non authenticated requests. The authenticated version will have context.user and token as definite fields

File: scripts/src/public/auth.ts (hunk 1 above)
Comment (doodyparizada) [reply scripts/src/public/auth.ts]: Maybe we should also check the expiration here. I thought `valid` checked that too

File: tests/src/services/users.spec.ts [hunk 4]
```diff
@@ -131,4 +131,19 @@ describe("api tests for /users", async () => {
...
+
+		await mock(app)
+			.delete("/v1/users/me/session")
+			.send()
+			.set("Authorization", `Bearer ${ token.id }`)
+			.expect(200);
+
+		token = (await AuthToken.findOne({ userId: user.id }))!;
+		expect(token.valid).toBeFalsy();
+	});
```
Comment (doodyparizada) [inline tests/src/services/users.spec.ts]: NiCe!

File: scripts/src/public/routes/users.ts (hunk 3 above)
Comment (nitzantomer) [reply scripts/src/public/routes/users.ts]: that's a good idea, but it's a task/PR of it's own

**Pull request merged**